packages/keystone/lib/Keystone/index.js

```diff
@@ -316,11 +330,77 @@ module.exports = class Keystone {
         _ksListsMeta: (_, args, context) =>
           this.listsArray.filter(list => list.access.read).map(list => list.listMeta(context)),
+        ...objMerge([
+          ...this.queries.map(({ schema, resolver }) => {
+            const queryName = gql(`type t { ${schema} }`).definitions[0].fields[0].name.value;
+            return {
+              [queryName]: (obj, args, context, info) =>
+                resolver(obj, args, context, info, {
+                  query: (queryString, { skipAccessControl = false, variables } = {}) => {
+                    let passThroughContext = context;
+
+                    if (skipAccessControl) {
+                      passThroughContext = {
+                        ...context,
+                        getListAccessControlForUser: () => true,
+                        getFieldAccessControlForUser: () => true,
+                      };
+                    }
+
+                    const graphQLQuery = this._graphQLQuery[context.schemaName];
+
+                    if (!graphQLQuery) {
+                      return Promise.reject(
+                        new Error(
+                          'No executable schema is available. Have you setup `@keystone-alpha/app-graphql`?'
+                        )
+                      );
+                    }
+
+                    return graphQLQuery(queryString, passThroughContext, variables);
+                  },
+                }),
+            };
+          }),
+        ]),
       },
 
       Mutation: {
         ...objMerge(firstClassLists.map(list => list.gqlAuxMutationResolvers)),
         ...objMerge(firstClassLists.map(list => list.gqlMutationResolvers)),
+        ...objMerge([
+          ...this._extendedMutations.map(({ schema, resolver }) => {
+            const mutationName = gql(`type t { ${schema} }`).definitions[0].fields[0].name.value;
+            return {
+              [mutationName]: (obj, args, context, info) =>
+                resolver(obj, args, context, info, {
+                  query: (queryString, { skipAccessControl = false, variables } = {}) => {
+                    let passThroughContext = context;
+
+                    if (skipAccessControl) {
+                      passThroughContext = {
+                        ...context,
+                        getListAccessControlForUser: () => true,
+                        getFieldAccessControlForUser: () => true,
+                      };
+                    }
+
+                    const graphQLQuery = this._graphQLQuery[context.schemaName];
+
+                    if (!graphQLQuery) {
+                      return Promise.reject(
+                        new Error(
+                          'No executable schema is available. Have you setup `@keystone-alpha/app-graphql`?'
+                        )
+                      );
+                    }
+
+                    return graphQLQuery(queryString, passThroughContext, variables);
+                  },
+                }),
```

Comment:
Some dupe code here ☝️ can we factor that out?

Reply:
Yep, this is also duplicated in `List/index.js`, so I just need to work out the right place to factor this out to 👍 Wanted to get the API 👍 from everyone before I spent too long on polish :-)